ui/admin/app/components/webhooks/WebhookForm.tsx

```diff
@@ -0,0 +1,179 @@
+
+            {/* Extract to custom github component */}
+
+            <ControlledCustomInput
+                control={form.control}
+                name="workflow"
+                label="Workflow"
+                description="The workflow that will be triggered when the webhook is called."
+            >
+                {({ field: { ref: _, ...field }, className }) => (
+                    <Select {...field} onValueChange={field.onChange}>
+                        <SelectTrigger className={className}>
+                            <SelectValue placeholder="Select a workflow" />
+                        </SelectTrigger>
+
+                        <SelectContent>{getWorkflowOptions()}</SelectContent>
+                    </Select>
+                )}
+            </ControlledCustomInput>
+
+            <ControlledInput
+                control={form.control}
+                name="secret"
+                label="Secret"
+                description="This secret should match the secret you provide to GitHub."
+                placeholder={hasSecret ? "(unchanged)" : ""}
+            />
+
+            <ControlledInput
+                control={form.control}
+                name="token"
+                label="Token (optional)"
+                description="Optionally provide a token to add an extra layer of security."
+                placeholder={hasToken ? "(unchanged)" : ""}
+            />
+
+            <ControlledCustomInput
+                control={form.control}
+                name="headers"
+                label="Headers"
+            >
+                {({ field }) => (
+                    <MultiSelect
+                        {...field}
+                        options={GithubHeaderOptions}
+                        value={field.value.map((v) => ({ label: v, value: v }))}
+                        creatable
+                        onChange={(value) =>
+                            field.onChange(value.map((v) => v.value))
+                        }
+                        side="top"
+                    />
+                )}
+            </ControlledCustomInput>
+
+            <Button
+                className="w-full"
+                type="submit"
+                disabled={isLoading}
+                loading={isLoading}
+            >
+                {isEdit ? "Update Webhook" : "Create Webhook"}
+            </Button>
+        </form>
+    );
+
+    function getWorkflowOptions() {
+        if (getWorkflows.isLoading)
+            return (
+                <SelectEmptyItem disabled>Loading workflows...</SelectEmptyItem>
+            );
+
+        if (!workflows?.length)
+            return (
+                <SelectEmptyItem disabled>No workflows found</SelectEmptyItem>
+            );
+
+        return workflows.map((workflow) => (
+            <SelectItem key={workflow.id} value={workflow.id}>
+                {workflow.name}
+            </SelectItem>
+        ));
+    }
+}
+
+const GithubHeaderOptions = [
```

Comment:
Do we allow any header? these headers seems to be specific to github

Reply:
This form will be adapted in the future to switch between various providers. For now it's specifically for github